src/Common/tests/Collections/IListTest.cs

```diff
@@ -0,0 +1,1692 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Linq;
+using System.Collections;
+using System.Collections.Generic;
+using Xunit;
+// ReSharper disable InconsistentNaming
+
+namespace Tests.Collections
+{
+    public enum CollectionOrder
```

Comment:
While we don't have a written rule that I can find of "one namespace-scoped type per file", I'm not sure that I'd look in IListTest for CollectionOrder, given that it's consumed by the IEnumerableTest class.

One type per file seems like a good starting point.